perception/image_projection_based_fusion/src/roi_detected_object_fusion/node.cpp

```diff
@@ -0,0 +1,179 @@
+      transformPoints(vertices, object2camera_affine, vertices_camera_coord);
+    }
+
+    double min_x(std::numeric_limits<double>::max()), min_y(std::numeric_limits<double>::max()),
+      max_x(std::numeric_limits<double>::min()), max_y(std::numeric_limits<double>::min());
+    std::size_t point_on_image_cnt = 0;
+    for (const auto & point : vertices_camera_coord) {
+      if (point.z() <= 0.0) {
+        continue;
+      }
+
+      Eigen::Vector2d proj_point;
+      {
+        Eigen::Vector4d proj_point_hom =
+          camera_projection * Eigen::Vector4d(point.x(), point.y(), point.z(), 1.0);
+        proj_point = Eigen::Vector2d(
+          proj_point_hom.x() / (proj_point_hom.z()), proj_point_hom.y() / (proj_point_hom.z()));
+      }
+
+      min_x = std::min(proj_point.x(), min_x);
+      min_y = std::min(proj_point.y(), min_y);
+      max_x = std::max(proj_point.x(), max_x);
+      max_y = std::max(proj_point.y(), max_y);
+
+      if (
+        proj_point.x() >= 0 && proj_point.x() <= image_width - 1 && proj_point.y() >= 0 &&
+        proj_point.y() <= image_height - 1) {
+        point_on_image_cnt++;
+
+        if (debugger_) {
+          debugger_->obstacle_points_.push_back(proj_point);
+        }
+      }
+    }
+    if (point_on_image_cnt == 0) {
+      continue;
+    }
+
+    min_x = std::max(min_x, 0.0);
+    min_y = std::max(min_y, 0.0);
+    max_x = std::min(max_x, image_width - 1);
+    max_y = std::min(max_y, image_height - 1);
+
+    // build roi
+    sensor_msgs::msg::RegionOfInterest roi;
+    roi.x_offset = static_cast<std::uint32_t>(min_x);
+    roi.y_offset = static_cast<std::uint32_t>(min_y);
+    roi.width = static_cast<std::uint32_t>(max_x) - static_cast<std::uint32_t>(min_x);
+    roi.height = static_cast<std::uint32_t>(max_y) - static_cast<std::uint32_t>(min_y);
+    object_roi_map.insert(std::make_pair(obj_i, roi));
+
+    if (debugger_) {
+      debugger_->obstacle_rois_.push_back(roi);
+    }
+  }
+}
+
+void RoiDetectedObjectFusionNode::updateDetectedObjectClassification(
+  const std::vector<DetectedObjectWithFeature> & image_rois,
+  const std::map<std::size_t, sensor_msgs::msg::RegionOfInterest> & object_roi_map,
+  std::vector<DetectedObject> & output_objects)
+{
+  for (std::size_t roi_i = 0; roi_i < image_rois.size(); roi_i++) {
+    const auto & roi = image_rois.at(roi_i).feature.roi;
+    std::size_t object_i = 0;
+    double max_iou = 0.0;
+    for (auto object_itr = object_roi_map.begin(); object_itr != object_roi_map.end();
+         object_itr++) {
+      double iou(0.0), iou_x(0.0), iou_y(0.0);
+      if (use_iou_) {
+        iou = calcIoU(object_itr->second, roi);
+      }
+      if (use_iou_x_) {
+        iou_x = calcIoUX(object_itr->second, roi);
+      }
+      if (use_iou_y_) {
+        iou_y = calcIoUY(object_itr->second, roi);
+      }
+
+      if (iou + iou_x + iou_y > max_iou) {
+        object_i = object_itr->first;
```

Comment:
[IMO] Using range-based for statement w/ const reference.
```suggestion
    for (const auto & obj_map : object_roi_map) {
      double iou(0.0), iou_x(0.0), iou_y(0.0);
      if (use_iou_) {
        iou = calcIoU(obj_map.second, roi);
      }
      if (use_iou_x_) {
        iou_x = calcIoUX(obj_map.second, roi);
      }
      if (use_iou_y_) {
        iou_y = calcIoUY(obj_map.second, roi);
      }

      if (iou + iou_x + iou_y > max_iou) {
        object_i = obj_map.first;

```

Reply:
changed at [a33f315](https://github.com/autowarefoundation/autoware.universe/pull/548/commits/a33f315f5c05d6d651483cb2c376d8eaa865897a).